Cargo.toml

```diff
@@ -16,33 +16,44 @@ description = "Submit extrinsics (transactions) to a substrate node via RPC"
 
+[features]
+client = ["substrate-subxt-client"]
+
 [dependencies]
-log = "0.4"
-thiserror = "1.0"
+log = "0.4.8"
+thiserror = "1.0.19"
 futures = "0.3.5"
-jsonrpsee = { version = "0.1", features = ["ws"] }
-num-traits = { version = "0.2", default-features = false }
-serde = { version = "1.0", features = ["derive"] }
-serde_json = "1.0"
-url = "2.1"
+jsonrpsee = { version = "0.1.0", features = ["ws"] }
+num-traits = { version = "0.2.11", default-features = false }
+serde = { version = "1.0.111", features = ["derive"] }
+serde_json = "1.0.53"
+url = "2.1.1"
 codec = { package = "parity-scale-codec", version = "1.3", default-features = false, features = ["derive", "full"] }
 
-frame-metadata = { version = "11.0.0-rc2", package = "frame-metadata" }
-frame-support = { version = "2.0.0-rc2", package = "frame-support" }
-sp-runtime = { version = "2.0.0-rc2", package = "sp-runtime" }
-sp-version = { version = "2.0.0-rc2", package = "sp-version" }
-pallet-indices = { version = "2.0.0-rc2", package = "pallet-indices" }
-hex = "0.4.0"
-sp-rpc = { version = "2.0.0-rc2", package = "sp-rpc" }
-sp-core = { version = "2.0.0-rc2", package = "sp-core" }
-sc-rpc-api = { version = "0.8.0-rc2", package = "sc-rpc-api" }
-sp-transaction-pool = { version = "2.0.0-rc2", package = "sp-transaction-pool" }
+frame-metadata = { version = "11.0.0-rc3", package = "frame-metadata" }
+frame-support = { version = "2.0.0-rc3", package = "frame-support" }
+sp-runtime = { version = "2.0.0-rc3", package = "sp-runtime" }
+sp-version = { version = "2.0.0-rc3", package = "sp-version" }
+pallet-indices = { version = "2.0.0-rc3", package = "pallet-indices" }
+hex = "0.4.2"
+sp-rpc = { version = "2.0.0-rc3", package = "sp-rpc" }
+sp-core = { version = "2.0.0-rc3", package = "sp-core" }
+sc-rpc-api = { version = "0.8.0-rc3", package = "sc-rpc-api" }
+sp-transaction-pool = { version = "2.0.0-rc3", package = "sp-transaction-pool" }
+substrate-subxt-client = { path = "client", optional = true }
 substrate-subxt-proc-macro = { version = "0.8.0", path = "proc-macro" }
 
 [dev-dependencies]
-async-std = { version = "1.5.0", features = ["attributes"] }
-env_logger = "0.7"
-wabt = "0.9"
-frame-system = { version = "2.0.0-rc2", package = "frame-system" }
-pallet-balances = { version = "2.0.0-rc2", package = "pallet-balances" }
-sp-keyring = { version = "2.0.0-rc2", package = "sp-keyring" }
+async-std = { version = "=1.5.0", features = ["attributes"] }
+env_logger = "0.7.1"
+wabt = "0.9.2"
+frame-system = { version = "2.0.0-rc3", package = "frame-system" }
+node-template = { git = "https://github.com/paritytech/substrate" }
+pallet-balances = { version = "2.0.0-rc3", package = "pallet-balances" }
+sp-keyring = { version = "2.0.0-rc3", package = "sp-keyring" }
+substrate-subxt-client = { path = "client" }
+tempdir = "0.3.7"
+
+[patch.crates-io]
+sc-network = { git = "https://github.com/paritytech/substrate" }
+sc-service = { git = "https://github.com/paritytech/substrate" }
```

Comment:
What are these patches for? Something not yet included in `rc3`? It might prevent publishing the main crate

Reply:
The problem is that node template isn't published, but it's required for the tests. So this makes sure that when it's checked out locally it uses the same versions as the template. I was hoping this trick would allow publish to crates.io